plugins/state_history_plugin/state_history_plugin.cpp

```diff
@@ -99,6 +99,7 @@ struct state_history_plugin_impl : std::enable_shared_from_this<state_history_pl
    std::set<acceptor_type>          acceptors;
 
    named_thread_pool<struct ship> thread_pool;
+   bool                           thread_pool_started = false;
```

Comment:
As you have it this needs to be an `std::atomic<bool>` as it is accessed from two different threads. You could call it `plugin_started` and set it in `state_history_plugin::plugin_startup` and then it would only be accessed from the main thread.

Reply:
Thank you, that looks better, done.